api/krusty/repeatbase_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+// Here is a structure of a kustomization of one resource inheriting from
+// two bases. One of those bases is shared betwen the canary base and the
+// final resource. This is named canary as it is a simple pattern to
+// duplicate a resource that can be used with canary deployments.
+//
+// deployment-base
+// |    deployment.yaml
+// |    kustomization.yaml
+// deployment-canary
+// |    deployment-canary-patch.yaml
+// |    kustomization.yaml
+// deployment-a
+// |    deployment-a-patch.yaml
+// |    kustomization.yaml
+func TestRepeatBase(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK("/app/deployment-base", `
+resources:
+  - deployment.yaml
+`)
+	th.WriteF("/app/deployment-base/deployment.yaml", `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deployment
+spec:
+  selector:
+    matchLabels:
+      component: deployment
+  template:
+    metadata:
+      labels:
+        component: deployment
+    spec:
+      containers:
+      - name: container-a
+        image: image
+`)
+
+	th.WriteK("/app/deployment-canary", `
+resources:
+  - ../deployment-base
+patches:
+- patch: |
+    - op: replace
+      path: /metadata/name
+      value: deployment-canary
+  target: 
+    kind: Deployment
+- path: deployment-canary-patch.yaml
+`)
+	th.WriteF("/app/deployment-canary/deployment-canary-patch.yaml", `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deployment-canary
+  labels:
+    type: canary
+spec:
+  selector:
+    matchLabels:
+      component: deployment
+      type: canary
+  template:
+    metadata:
+      labels:
+        component: deployment
+    spec:
+      containers:
+      - name: container-a
+        image: image-canary
+`)
+
+	th.WriteK("/app/deployment-a", `
+nameSuffix: -a
+resources:
+  - ../deployment-base
+  - ../deployment-canary
+patches:
+- path: deployment-a-base-patch.yaml
+- path: deployment-a-canary-patch.yaml
+`)
+  th.WriteF("/app/deployment-a/deployment-a-base-patch.yaml", `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: container-a
+        image: image-a
+`)
+	th.WriteF("/app/deployment-a/deployment-a-canary-patch.yaml", `
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deployment-canary
+spec:
+  template:
+    spec:
+      containers:
+      - name: container-a
+        image: image-canary-a
+`)
+
+	th.WriteK("/app", `
+resources:
+  - deployment-a
+`)
+
+	m := th.Run("/app", th.MakeDefaultOptions())
```

Comment:
In #3609 you reported that this test will produce an error (and thus will fail here); could you please change this test to 
1) include a comment that it is currently broken
2) use th.RunWithErr instead, such that the output if this test matches what the current output actually is (e.g. the test should "pass" even though it is broken behavior"
3) include a comment of what the expected output should be - or include what the output should be as a comment in the issue you filed?

Reply:
@natasha41575 Thanks for the comments. Hopefully I've addressed all items.

Reply:
@chicocvenancio awesome, thanks. last couple of things: could you 1) include a link to the issue you filed in the comment documenting that this is broken behavior and 2) squash this PR down to 1 commit?

Reply:
Done!